test/k8sT/istio.go

```diff
@@ -0,0 +1,276 @@
+			bookinfoV1YAML := helpers.ManifestGet("bookinfo-v1-istio.yaml")
+			bookinfoV2YAML := helpers.ManifestGet("bookinfo-v2-istio.yaml")
+
+			// This YAML is copied from examples/kubernetes-istio.
+			initPolicyPath := helpers.ManifestGet("istio-sidecar-init-policy.yaml")
+			l7PolicyPath := helpers.ManifestGet("cnp-specs.yaml")
+
+			// Create the L7 policy before creating the pods, in order to test
+			// that the sidecar proxy mode doesn't deadlock on endpoint
+			// creation in this case.
+			// Also create an init policy that allows traffic to Istio services
+			// to ease pod creation.
+			policyPaths = []string{initPolicyPath, l7PolicyPath}
+			for _, policyPath := range policyPaths {
+				By("Creating policy in file %q", policyPath)
+				_, err := kubectl.CiliumPolicyAction(helpers.KubeSystemNamespace, policyPath, helpers.KubectlApply, helpers.HelperTimeout)
+				Expect(err).Should(BeNil(), "Unable to create policy %q", policyPath)
+			}
+
+			resourceYAMLPaths = []string{bookinfoV1YAML, bookinfoV2YAML}
+			for _, resourcePath := range resourceYAMLPaths {
+				By("Creating resources in file %q", resourcePath)
+				res := kubectl.Create(resourcePath)
+				res.ExpectSuccess("Unable to create resource %q: %s", resourcePath, res.CombineOutput())
+			}
+		})
+
+		AfterEach(func() {
+			for _, resourcePath := range resourceYAMLPaths {
+				By("Deleting resource in file %q", resourcePath)
+				// Explicitly do not check result to avoid having assertions in AfterEach.
+				_ = kubectl.Delete(resourcePath)
+			}
+
+			for _, policyPath := range policyPaths {
+				By("Deleting policy in file %q", policyPath)
+				// Explicitly do not check result to avoid having assertions in AfterEach.
+				_ = kubectl.Delete(policyPath)
+			}
+		})
+
+		// shouldConnect checks that srcPod can connect to dstURI.
+		shouldConnect := func(srcPod, dstURI string) bool {
+			By("Checking that %q can connect to %q", srcPod, dstURI)
+			res := kubectl.ExecPodCmd(
+				helpers.DefaultNamespace, srcPod, fmt.Sprintf("%s %s", wgetCommand, dstURI))
+			if !res.WasSuccessful() {
+				GinkgoPrint("Unable to connect from %q to %q: %s", srcPod, dstURI, res.CombineOutput())
+				return false
+			}
+			return true
+		}
+
+		// shouldNotConnect checks that srcPod cannot connect to dstURI.
+		shouldNotConnect := func(srcPod, dstURI string) bool {
+			By("Checking that %q cannot connect to %q", srcPod, dstURI)
+			res := kubectl.ExecPodCmd(
+				helpers.DefaultNamespace, srcPod, fmt.Sprintf("%s %s", wgetCommand, dstURI))
+			if res.WasSuccessful() {
+				GinkgoPrint("Was able to connect from %q to %q, but expected no connection: %s", srcPod, dstURI, res.CombineOutput())
+				return false
+			}
+			return true
+		}
+
+		// formatLabelArgument formats the provided key-value pairs as labels for use in
+		// querying Kubernetes.
+		formatLabelArgument := func(firstKey, firstValue string, nextLabels ...string) string {
```

Comment:
don't really like this function (I hate it from the services), what if you use a map here
```golang
func(entries map[string]string) string{
	var result string = ""
	for k,v := range entries {
		result += fmt.Sprintf("%s=%s")
	}
	return result
}
```

Reply:
What do you not like about this function exactly?

Reply:
It's not self-explicit when it's used. IMHO